frontend/web/components/modals/InviteUsers.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useState, useEffect, useRef, FC } from 'react'
+import Button from 'components/base/forms/Button'
+import ConfigProvider from 'common/providers/ConfigProvider'
+import Constants from 'common/constants'
+import Icon from 'components/Icon'
+import { add } from 'ionicons/icons'
+import { IonIcon } from '@ionic/react'
+import { getPlanBasedOption } from 'components/PlanBasedAccess'
+import InputGroup from 'components/base/forms/InputGroup'
+import OrganisationProvider from 'common/providers/OrganisationProvider'
+import Utils from 'common/utils/utils'
+import _ from 'lodash'
+import AppActions from 'common/dispatcher/app-actions'
+import ErrorMessage from 'components/ErrorMessage'
+import AccountStore from 'common/stores/account-store'
+import { close as closeIcon } from 'ionicons/icons'
+import { MultiValueProps } from 'react-select/lib/components/MultiValue'
+import { useGetGroupsQuery } from 'common/services/useGroup'
+
+interface Invite {
+  temporaryId: string
+  emailAddress?: string
+  role?: string
+  groups?: number[]
+}
+
+const CustomMultiValue = (props: MultiValueProps<GroupOption>) => {
+  const { data, removeProps } = props
+
+  if (!data?.label || !data?.value) {
+    return null
+  }
+
+  return (
+    <Row className='chip p-1'>
+      <span className='font-weight-bold'>{data.label}</span>
+      {
+        <span className='chip-icon ion'>
+          <IonIcon
+            icon={closeIcon}
+            onClick={() => removeProps?.onClick?.(data)}
+          />
+        </span>
+      }
+    </Row>
+  )
+}
+
+type GroupOption = { label: string; value: number }
+
+const InviteUsers: FC = () => {
+  const [invites, setInvites] = useState<Invite[]>([
+    { temporaryId: Utils.GUID() },
+  ])
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const { id: orgId } = AccountStore.getOrganisation()
+
+  const { data: groupData } = useGetGroupsQuery({ orgId: orgId, page: 1 })
+
+  const groupOptions = groupData?.results.map((group) => ({
+    label: group.name,
+    value: group.id,
+  }))
+
+  useEffect(() => {
+    const focusTimeout = setTimeout(() => {
+      inputRef.current?.focus()
+    }, 500)
+
+    return () => {
+      clearTimeout(focusTimeout)
+    }
+  }, [])
+
+  const isValid = (): boolean => {
+    return _.every(
+      invites,
+      (invite) => Utils.isValidEmail(invite.emailAddress) && invite.role,
+    )
+  }
+
+  const onChange = (
+    id: string,
+    key: keyof Invite,
+    value: string | number[],
+  ): void => {
+    setInvites((prev) =>
+      prev.map((invite) => {
+        if (invite.temporaryId === id) {
+          return {
+            ...invite,
+            [key]: value,
+          }
+        }
+        return invite
+      }),
+    )
+  }
+
+  const deleteInvite = (id: string): void => {
+    setInvites(invites.filter((invite) => invite.temporaryId !== id))
+  }
+
+  const close = (invites: Omit<Invite, 'temporaryId'>[]): void => {
+    AppActions.inviteUsers(invites)
```

Comment:
I think this is a good candidate for migration given this is in tsx now, all the invite actions